packages/ts/src/rules/explicitModuleBoundaryTypes.ts

```diff
@@ -0,0 +1,171 @@
+// eslint-disable perfectionist/sort-switch-case -- cases are ordered by importance, not alphabetically
+import type { ReportMessageData } from "@flint.fyi/core";
+import type { RuleContext } from "@flint.fyi/core/src/types/context.js";
+
+import * as ts from "typescript";
+
+import { getTSNodeRange } from "../getTSNodeRange.js";
+import { typescriptLanguage } from "../language.js";
+
+const messages = {
+	fnParam: {
+		primary:
+			"Public facing functions should have explicit types for all parameters.",
+		secondary: ["{{ parameterName }} is declared here."],
+		suggestions: [],
+	},
+	fnReturn: {
+		primary: "Public facing functions should have an explicit return type.",
+		secondary: [],
+		suggestions: [],
+	},
+	fnReturnWithSuggestion: {
+		primary: "Public facing functions should have an explicit return type.",
+		secondary: [],
+		suggestions: [
+			"Add `{{ returnType }}` to the return type of this function.",
```

Comment:
Directly putting `typeToString`-ified types into error messages makes them super long when the type isn't straightforward. I think it's easier to just go with a general _"return type"_:

```suggestion
			"Add an explicit return type to the return type of this function.",
```

This is a practice we've roughly been doing in typescript-eslint but haven't really formalized. Filed for discussion here at least: _(cross-linking)_